File: viajerosrepojava-master/viajerosrepojava-master/viajeros/viajeros/src/main/java/ps/com/viajeros/dtos/viaje/incidentes/IncidenteDto.java
package ps.com.viajeros.dtos.viaje.incidentes;

import lombok.*;
import ps.com.viajeros.entities.viajes.EstadoResolucion;
import ps.com.viajeros.entities.viajes.TipoIncidente;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class IncidenteDto {
    private Long idIncidente;
    private Long viajeId;
    private String descripcion;
    private TipoIncidente tipoIncidente;
    private LocalDateTime fechaIncidente;
    private Boolean isPasajero;
    private Long denunciadoId;
    private Long reportadoPorId;
    private EstadoResolucion estadoResolucion;
    private String resolucion;
    private LocalDateTime fechaResolucion;
}

File: viajerosrepoangular-master/viajerosrepoangular-master/viajeros/src/app/models/Viajes/IncidenteDto.ts
export interface IncidenteDto {
    idIncidente: number;
    viajeId: number;
    descripcion: string;
    tipoIncidente: TipoIncidente;
    fechaIncidente: Date;
    isPasajero: boolean;           // Indica si el denunciado es pasajero
    denunciadoId: number;          // ID del usuario denunciado
    reportadoPorId: number;        // ID del usuario que reporta el incidente
    estadoResolucion: EstadoResolucion;
    resolucion?: string;           // opcional, puede que no esté resuelto
    fechaResolucion?: Date;        // opcional, la fecha de resolución
  }
  
  export enum TipoIncidente {
    MALTRATO_CHOFER = 'MALTRATO_CHOFER',
    MALTRATO_PASAJERO = 'MALTRATO_PASAJERO',
    SUCIEDAD = 'SUCIEDAD',
    PROBLEMA_APP = 'PROBLEMA_APP',
    OTRO = 'OTRO'
  }
  
  export enum EstadoResolucion {
    PENDIENTE = 'PENDIENTE',
    RESUELTO = 'RESUELTO',
    NO_RESUELTO = 'NO_RESUELTO'
  }
  